### DISCUSSION TITLE
istio components status not found istio-ingressgateway

### DISCUSSION THREAD

**Discussion by dimitarshenkov** [kiali/kiali#5381] [General]
Hi , we have Kiali installed with Helm.
also istio with private and public gateways 
Getting ERR in the UI
![image](https://user-images.githubusercontent.com/99493924/184172784-28e9fc8e-8b74-4dfd-b828-b64ccf3ddcb6.png)
CRD has
![image](https://user-images.githubusercontent.com/99493924/184172897-b309c6df-453f-43db-9c73-e624e5e906b9.png)
Default app labels for the ingresses are changed to those ones

**Comment by jmazzitelli** [comment]
[converting to discussion, as this is likely not a bug but just a user question on how to configure kiali]

**Comment by jmazzitelli** [comment]
The answer to this question likely requires us to know what version of Kiali you are using. You  did not say what version of Kiali you are using. What is the helm command you used to install?

Also, note: you say "CRD has" (which I assume you mean the Kiali CR, not the CRD)... the Kiali CR is only used if you installed and are using the Kiali operator. Did you install the operator using the operator helm chart? Or did you install Kiali using the server helm chart (hence, why I ask, "What is the command you used to install Kiali").

So we need to know:
1. What command did you use to install Kiali
2. Are you using the operator?
3. What version of Kiali?

I suspect the answer to 1. will provide the answer to 2. and 3.

**Reply by dimitarshenkov** [reply]
> 

![image](https://user-images.githubusercontent.com/99493924/184302374-6623172f-5296-4e41-a1e4-204f248dcb27.png)
Hi @jmazzitelli its Installed by operator and Terraform
```
resource "kubectl_manifest" "kiali-cr" {
  yaml_body  = <<YAML
apiVersion: kiali.io/v1alpha1
kind: Kiali
metadata:
  annotations:
    meta.helm.sh/release-name: kiali-operator
    meta.helm.sh/release-namespace: kiali-operator
    ansible.operator-sdk/verbosity: "1"
  generation: 7
  labels:
    app: kiali-operator
    app.kubernetes.io/instance: kiali-operator
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: kiali-operator
    app.kubernetes.io/part-of: kiali-operator
    # app.kubernetes.io/version: v1.30.0
    # helm.sh/chart: kiali-operator-1.30.0
    # version: v1.30.0
  name: kiali
  namespace: istio-system
  selfLink: /apis/kiali.io/v1alpha1/namespaces/istio-system/kialis/kiali
spec:
  istio_namespace: istio-system
  deployment:
    namespace: istio-system
    accessible_namespaces:
    - '**'
    ingress_enabled: false
  external_services:
    grafana:
      url: "https://${var.cluster_name_short}grafana.mansion.gi"
      dashboards:
      - name: Istio Service Dashboard
        variables:
          namespace: var-namespace
          service: var-service
      - name: Istio Workload Dashboard
        variables:
          namespace: var-namespace
          workload: var-workload
      - name: Istio Control Plane Dashboard
      - name: Istio Mesh Dashboard
      - name: Istio Performance Dashboard      
    tracing:
      enabled: true
      in_cluster_url: http://tracing.istio-system/jaeger
      url: "https://${var.cluster_name_short}jaeger.mansion.gi/jaeger"
      use_grpc: false 
    prometheus:
      url: http://prometheus-operated.monitoring:9090
  server:
    web_fqdn: "${var.cluster_name_short}kiali.mansion.gi"
    web_root: "/"
    web_schema: https
  #https://kiali.io/docs/faq/istio-component-status/#one-component-is-not-found-but-i-can-see-it-running-what-can-i-do
    istio:
      component_status: 
        components:
        - app_label: "istio-ingressgateway-public"
          is_core: true
          is_proxy: true
          namespace: istio-system
        - app_label: "istio-ingressgateway-private"
          is_core: true
          is_proxy: true
          namespace: istio-system

YAML
  depends_on = [helm_release.kiali-operator]
}
```

**Reply by jmazzitelli** [reply]
Your Kiali CR is not correct. You set `spec.server.istio.component_status` which is wrong - from your output:
```
spec:
...
  server:
...
    istio:
      component_status: 
...
```

The above is invalid.

Please follow the documentation. Read the Kiali CR reference for the sections that are relevant for your question here: https://kiali.io/docs/configuration/kialis.kiali.io/#.spec.external_services.istio.component_status

I suggest you run the validation tool on your Kiali CR - it will tell you about any issues in your kiali CR. To run the validation tool, please follow the documentation here: https://kiali.io/docs/configuration/kialis.kiali.io/#validating-your-kiali-cr

**Reply by dimitarshenkov** [reply]
Thanks that was it.

**Comment by jmazzitelli** [comment]
Here are some docs about configuring component_status that may help: https://kiali.io/docs/faq/istio-component-status/

**Reply by dimitarshenkov** [reply]
This did not helping , I dont want to change label name  from default "app" to sth else, and it wont help any way
